portfolio/src/main/java/com/google/sps/servlets/ChartServlet.java

```diff
@@ -0,0 +1,54 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/chart")
+public class ChartServlet extends HttpServlet {
+
+  private Map<String, Integer> mapData = new HashMap<>();
+  private ArrayList<String> votedUsers = new ArrayList<>();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    MapAndUsers obj = new MapAndUsers(mapData, votedUsers);
+    response.setContentType("application/json");
+    String json = convertUserToJsonUsingGson(obj);
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String genre = request.getParameter("selGenre");
+    String userId = request.getParameter("postingUser");
+    int currentVotes = mapData.containsKey(genre) ? mapData.get(genre) : 0;
+    mapData.put(genre, currentVotes + 1);
```

Comment:
You should check if our current user has already voted before letting them vote a second time.